### PR TITLE
camera: exclude "photos taken with" categories

### PR DESCRIPTION
**Pull request by waldyrious**
Not sure if category exclusions are recursive. If they're not, this PR is pretty much useless, since almost all of the contents of [Category:Photographs by camera manufacturer](https://commons.wikimedia.org/wiki/Category:Photographs_by_camera_manufacturer) are in its subcategories.

Assuming this works, it would fix some instances of "depicts a camera" questions, like [File:View from Mount Seymour Trail.jpg](https://commons.wikimedia.org/wiki/File:View_from_Mount_Seymour_Trail.jpg), which is in [Category:Taken with Olympus Air A01](https://commons.wikimedia.org/wiki/Category:Taken_with_Olympus_Air_A01).

### CODE REVIEW

**Comment by addshore** [comment]
So the code looks down the tree from the top level.
If while navigating that tree `Category:Photographs by camera manufacturer` is encountered, the code will not continue to explore down that tree!

So, will this code have the desired effect?
Another alternative would be ignoring `Taken with` in the regex!

**Comment by waldyrious** [comment]
> So, will this code have the desired effect?

I'm not sure. I mean, there is clearly a path from the top-level to [Category:Olympus Air A01](https://commons.wikimedia.org/wiki/Category:Olympus_Air_A01) >  [Category:Taken with Olympus Air A01](https://commons.wikimedia.org/wiki/Category:Taken_with_Olympus_Air_A01) > [File:View from Mount Seymour Trail.jpg](https://commons.wikimedia.org/wiki/File:View_from_Mount_Seymour_Trail.jpg) without going through [Category:Photographs by camera manufacturer](https://commons.wikimedia.org/wiki/Category:Photographs_by_camera_manufacturer), so naively, I would suspect the restriction included here could be bypassed.

The `Taken with` regex might more reliable. Perhaps we would use both approaches?

**Comment by PiotrGackowski** [comment]
@addshore short description how to block categories and what is difference between excluded categories and regex will be nice - right now I dont know difference between them.

File: spec/depicts/misc/camera.yaml [hunk 1]
```diff
@@ -8,4 +8,5 @@ exclude:
   - Category:Camera parts
   - Category:Camera showrooms
   - Category:Camera accessories
-excludeRegex: "/(in art)/i"
+  - Category:Photographs by camera manufacturer
+excludeRegex: "/(in art|Taken with)/"
```
Comment (addshore) [inline spec/depicts/misc/camera.yaml:12]: should add the `/i` back so that it is case insensitive.

**Comment by addshore** [comment]
> @addshore short description how to block categories and what is difference between excluded categories and regex will be nice - right now I dont know difference between them.

Excluded categories must be an exact match with teh whole name of the category in order to skip it

So `Category:Foo` will only result in `Category:Foo` exactly being skipped.

However if we tweak the `excludeRegex` to include `foo` then any category with the word `foo` will be skipped.

You can test the regex easily in some online tools
https://www.regextester.com/
image
I can provide more links for reading on regex too :)
TLDR is that `/(foo|bar|baz)/i` means any category with `foo` or `bar` or `baz`

File: spec/depicts/misc/camera.yaml (hunk 1 above)
Comment (waldyrious) [reply spec/depicts/misc/camera.yaml:12]: It didn't feel necessary in this case, but sure, it's not likely that capitalization variations in these patterns would represent different intended meanings. I'll add it back.

**Review by addshore — APPROVED** [review]
